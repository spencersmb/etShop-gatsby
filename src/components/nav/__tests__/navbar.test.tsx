import React from 'react'
import renderer from 'react-test-renderer'
import Navbar from '../navbar'
import {
  render,
} from 'react-testing-library'

describe('Navbar Layout', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<Navbar />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('Should have Login and Home buttons', ( ) => {
    const { getByText, getByTestId } = render(<Navbar />)
    const element = getByTestId('navbar')
    // console.log('element', element.children[0].children.length)

    expect(getByText('Home')).toBeDefined()
    
  })
})