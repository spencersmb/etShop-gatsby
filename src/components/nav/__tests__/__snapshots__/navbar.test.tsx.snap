// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar Layout renders correctly 1`] = `
<nav
  className="sc-kAzzGY eDyPqo"
  data-testid="navbar"
>
  <div
    className="sc-kgoBCf fkElcL"
    data-testid="nav-logo"
  >
    <div
      className="sc-kGXeez debjTL"
    >
      <a
        href="/"
      >
        <test-file-stub />
      </a>
    </div>
  </div>
  <div
    className="sc-cSHVUG sc-jWBwVP edsDtl"
    onClick={[Function]}
  >
    <div
      className="sc-eHgmQL dgQcyp"
    >
      <test-file-stub />
    </div>
  </div>
  <div
    className="sc-chPdSV kvPeDg"
    data-testid="hamburger"
    onClick={[Function]}
  >
    Button
  </div>
  <div
    className="sc-kpOJdX htKNSJ"
  >
    <div
      className="sc-dxgOiQ fwiTqa"
      data-testid="nav-close"
      onClick={[Function]}
    >
      Close
    </div>
    <ul
      className="sc-ckVGcZ eTBjLu"
      data-testid="nav-center"
    >
      <li>
        <a
          href="/"
        >
          Products
        </a>
      </li>
      <li>
        <a
          href="https://every-tuesday.com"
          rel="noreferrer"
          target="_blank"
        >
          Blog
        </a>
      </li>
      <li>
        <a
          href="/support"
        >
          Support
        </a>
      </li>
    </ul>
    <div
      className="sc-jKJlTe cTeKkm"
    >
      <div
        className="sc-cSHVUG sc-eNQAEJ kASAKx"
      >
        <div
          className="sc-VigVT sc-kEYyzF iQlDtD"
          color="transparent"
          onClick={[Function]}
        >
          Sign In
        </div>
        <div
          className="sc-VigVT sc-hMqMXs hpwWkl"
          color="#A182E9"
          onClick={[Function]}
        >
          Join Now
        </div>
      </div>
      <div
        className="sc-cSHVUG sc-hSdWYo bzLIHN"
        onClick={[Function]}
      >
        <div
          className="sc-eHgmQL dgQcyp"
        >
          <test-file-stub />
        </div>
        <div
          className="sc-cvbbAY jrAWlO"
          data-testid="cart-count"
        >
          <span>
            
          </span>
        </div>
      </div>
    </div>
  </div>
</nav>
`;
